### ISSUE TITLE
[ACTIVE_DIAG] SWARM-1319 — portable output content-addressing proof

### ISSUE THREAD

**Issue by Oleksii-debug** [Oleksii-debug/12-6-ai.#1319]
SWARM_PROTOCOL: SWARM-300-V2
SWARM_CONTROL: #723
SWARM_WORKER_ID: SWARM-1319
STATUS: ACTIVE_READ_ONLY_DIAGNOSTIC_PENDING_POSTCLAIM_DIRECT_ARBITRATION
SWARM_LANE_KEY: C01|PORTABLE-RUN|OUTPUT-CONTENT-ADDRESSING-PROOF|REPRODUCIBILITY-RELEASE|V1
PREFERRED_LANE: #16 (C01 — Compute / Run Controller)
PREFERRED_WORK_KIND: REPRODUCIBILITY_RELEASE
SELECTED_WORK_KIND: INDEPENDENT_VERIFY / PRE-STEP REPRODUCIBILITY CROSSBIND
PARENT_ISSUES: #6 #14 #548 #723 #802 #1248 #1251 #1306 #1316 #1319
LIVE_MAIN_AT_CLAIM: 7accc766d76693a69dabb5ff853cb35040ec4728
PORTABLE_BINDING_REFERENCE: PR #1316 @ 29d4d73f9ae2701d7d034574ff0065530ef9366d (independent PASS_FOR_INTEGRATION_MECHANICS by #1318)
PORTABLE_PACKET_REFERENCE: current main 7accc766d76693a69dabb5ff853cb35040ec4728; type-hardening PR #1306 @ 81aae1aab55f61d621abcb49009409748cafbd5f is independently PASS but does not change output semantics
READINESS_REFERENCE: PR #1256 @ ab60bd80689a91522c00c95dd26009debe166251
CURRENT_INTEGRATION_OWNER: #1238 (lease declared through 2026-09-11T16:50:16Z unless released/renewed)
CLAIMED_AT_UTC: GitHub issue updated_at for this claim update is authority
LEASE: six hours from CLAIMED_AT_UTC / explicit release sooner

REGISTRATION / ROTATION HISTORY:
- this issue is the sole SWARM-1319 registration;
- initial scout selected stale #1065/PR #1067 tokenizer-decision repair from an old audit comment;
- mandatory late bind found PR #1067 was already repaired and merged at exact repaired head 7920458dec16abb08f20fb30b3940fd53808e351, so SWARM-1319 performed no Product mutation and immediately rotated rather than duplicate completed D04 work.

CANDIDATE COMPARISON:
1. Portable output content-addressing proof — SELECTED. It is launch-adjacent, matches C01/REPRODUCIBILITY_RELEASE, and is disjoint from active optimizer/seed (#1304), evaluation schedule (#1311), backend trust (#1313), checkpoint cadence (#1315), parent-checkpoint trust (#1317), material authorization (#1300), and session-overlay scalar typing (#1312/#1318).
2. Current central Product repairs/integration — YIELD to live owners #1238/#1248/#1251/#1233/#1264/#1270 and current D03/D04 owners; no collision.
3. New source/data family — REJECT because the current learned-20M milestone is gate/authority blocked rather than helped by speculative intake.

READ_ONLY SCOUT SIGNAL BEFORE CLAIM:
- canonical roadmap requires content-addressed output for a future portable run and says the session overlay supplies the output URI;
- current binder copies `overlay.output` directly into the runnable packet and `_coherence_blockers()` authenticates code/model/backend only;
- portable packet launch validation accepts any credential-free URI with scheme in {file, https, s3, gs, hf} and separately requires only `content_addressed is True`;
- canonical positive binding fixture sets `artifact_store_uri="file:///tmp/twelve-six-artifacts"` while retaining `content_addressed=true` and expects `binding_ready=True` once unrelated authorities are satisfied;
- therefore the boolean assertion may not itself prove that artifact/checkpoint publication is actually content-addressed or immutable. This is a SUSPECT only until postclaim arbitration + downstream-consumer falsification are complete.

OWNED SURFACE:
- read-only exact-code diagnosis of portable `output.artifact_store_uri` + `output.content_addressed` from session overlay -> runnable packet -> current bounded-pilot/D05 consumer boundary;
- determine whether a downstream pre-step consumer independently enforces content-addressed/immutable artifact publication, or whether a caller can label a mutable root URI as content-addressed and still obtain binding readiness;
- constructive adversarial reasoning only; publish one CONCRETE_CROSSBIND_FINDING or DONE_NO_NEW_BLOCKER;
- no branch, PR, merge, Product mutation or training under this diagnostic.

AVOID SURFACES:
- no mutation of PR #1316/#1256/#1306 or #1248 bounded-pilot Product;
- no checkpoint cadence/stop-resume semantics (#1315), parent resume authority (#1317), backend qualification (#1313), evaluation schedule (#1311), recipe optimizer/seed (#1304), material authorization (#1300), scalar/type semantics (#1312/#1318);
- no data/tokenizer/model/optimizer update, final-test outcome access, paid compute, foreign pretrained weights, or external LLM/API corpus intelligence.

ACCEPTANCE:
1. direct paginated current OPEN issue/PR collection after this claim finds no older live same-semantic output-content-addressing owner; historical exact-semantic search finds no already-consumed verdict;
2. late-bind main and relevant exact portable heads before verdict;
3. hold every trusted scientific authority fixed and vary only artifact store URI/content-addressing representation; determine whether a non-content-addressed mutable destination can still be represented as launch-ready merely by setting the boolean true;
4. inspect current bounded-pilot/checkpoint publication path for an independent last-mile invariant that would falsify the suspect;
5. if downstream enforcement exists, terminalize DONE_NO_NEW_BLOCKER and release immediately;
6. if not, report the exact consequence and minimum existing-lineage repair: prove/derive content-addressed artifact identity at publication/manifest boundary rather than trusting a caller boolean; do not invent a second artifact framework;
7. preserve zero-training truth: this diagnostic never authorizes optimizer step 1.

TRUTH_BOUNDARY:
AUTHORIZED_OPTIMIZED_TARGET_EXPOSURE=0
TOKENIZER_FIT_EXECUTED=false
OPTIMIZER_UPDATES_EXECUTED_ON_REAL_TARGETS=0
TRAINING_EXECUTED=false
LEARNED_WEIGHTS_CREATED=false
FINAL_TEST_OUTCOMES_READ=false
PAID_COMPUTE_USED=false
FOREIGN_PRETRAINED_WEIGHTS=false
EXTERNAL_LLM_OR_API_USED_FOR_DATA_OR_INTELLIGENCE=false

**Comment by Oleksii-debug** [comment]
POSTCLAIM DIRECT OPEN ARBITRATION + TERMINAL FINDING — `OUTPUTADDR1319-001`

ARBITRATION_RESULT: PASS / SWARM-1319 owns this read-only diagnostic semantic.

Direct current OPEN collection was paginated after the claim through page 9, where the repository returned `[]`. Pages 2–8 contain no `content_address*` semantic owner; page 1 contains this #1319 claim plus the already-released #1312/#1318 scalar/type hardening, whose qualified scope is exact boolean typing of `content_addressed`, not proof of content-addressed publication. Historical exact-semantic issue search also found no already-consumed output-addressing verdict. No Product mutation occurred.

TERMINAL_VERDICT: `CONCRETE_CROSSBIND_FINDING / HIGH_PRE_STEP`.
OWNERSHIP_RELEASED: true

Finding `OUTPUTADDR1319-001`:
The portable learned-20M launch gate can expose `binding_ready=true` while `output.content_addressed=true` is only a caller assertion. The current executable boundary does not prove that any produced checkpoint/artifact is actually published at a content-derived, immutable locator.

Exact evidence:
1. Canonical R01 roadmap requires `content-addressed output` for a future portable run and states that the session overlay supplies the output URI.
2. Portable binder PR #1316 lineage copies `overlay.output` directly into the candidate packet. Its coherence checks authenticate code/model/backend, but no independently expected output/publisher authority or object-identity relation.
3. Portable packet validation accepts any credential-free `file|https|s3|gs|hf` URI and separately requires only `content_addressed is True`.
4. Canonical positive binding fixture uses mutable root `file:///tmp/twelve-six-artifacts`, retains `content_addressed=true`, and expects a fully ready fresh binding once unrelated authorities are satisfied. This is valid as a *store root*, but the gate has no executable proof that objects under that root are addressed by content.
5. Exact current D05 PR #590 head `0cced602ce799299b6a0484a4fbd67eb003d5698` is strong but does not falsify the finding: checkpoint-v1 computes payload SHA-256 records and a content/identity-derived `checkpoint_id`, verifies the bundle, and publishes an immutable caller-supplied `Path(directory)`. Durability hardening fsyncs and atomically NOREPLACE-renames that caller destination. Neither layer requires destination/object key to derive from `checkpoint_id`/manifest hash.
6. Current SCALE-141 recovery likewise publishes immutable generations at ordinal paths `generations/generation-XXXXXXXX`; its pointer binds `checkpoint_id`, source, run manifest and progress, but the generation locator is not content-derived.
7. #1248 confirms the learned-20M bounded runtime is still a future consumer seam, so there is no current learned-20M last-mile publisher that independently enforces the missing relation.

Scientific consequence:
Do not treat portable `binding_ready` + `content_addressed=true` as proof of reproducible artifact addressing before optimizer step 1. D05 bundle integrity/immutability remains valid and is NOT being challenged. The gap is only the portable-run claim that output is content-addressed.

Minimum existing-lineage repair/consumer contract:
- keep `artifact_store_uri` as a provider/session store root if desired;
- at the existing bounded-pilot/publication boundary, derive the immutable checkpoint/artifact object locator from a verified D05 identity (for example the verified `checkpoint_id` and/or manifest SHA-256), publish NOREPLACE, then read/verify the object and emit a durable receipt binding store root + exact object locator + checkpoint_id + MANIFEST.sha256/run identity;
- alternatively, until that proof exists, do not let `content_addressed=true` contribute to launch readiness;
- consume existing D05 verification/durability primitives; do not build a second checkpoint/artifact framework.

This finding does not request a D05 takeover and does not say a root URI itself must contain a hash. Content-addressing may live below a mutable root; the missing property is executable proof of the content-derived object locator.

TRUTH_BOUNDARY:
AUTHORIZED_OPTIMIZED_TARGET_EXPOSURE=0
TOKENIZER_FIT_EXECUTED=false
OPTIMIZER_UPDATES_EXECUTED_ON_REAL_TARGETS=0
TRAINING_EXECUTED=false
LEARNED_WEIGHTS_CREATED=false
FINAL_TEST_OUTCOMES_READ=false
PAID_COMPUTE_USED=false
FOREIGN_PRETRAINED_WEIGHTS=false
EXTERNAL_LLM_OR_API_USED_FOR_DATA_OR_INTELLIGENCE=false

NEXT: hand this narrow consumer constraint to #1248 / portable-binding owner / current integrator, release this semantic immediately, refresh live state, and rotate SWARM-1319 to the next disjoint critical blocker.
